### PR TITLE
[kubernetes] Create basic outline for Kubernetes

### PR DESCRIPTION
**Pull request by TylerHelmuth**
This PR creates a basic structure for new Kubernetes documentation.

Related to:

- #3005

**Preview**: https://deploy-preview-3027--opentelemetry.netlify.app/docs/kubernetes/

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

**Review by cartermp — APPROVED** [review]
I'm a fan. Will let @chalin and @svrnm give this a lookie-loo too -- we'll iterate a bit as we go, but this seems like a good overall layout to start.

File: content/en/docs/kubernetes/getting-started.md
Comment (svrnm) [inline content/en/docs/kubernetes/getting-started.md:1]: I am not a big fan of "coming soon" pages, they add no value for the reader. I suggest to delete this file for this PR

File: content/en/docs/kubernetes/helm.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+title: OpenTelemetry Helm Charts
+weight: 3
+spelling: cSpell:ignore Otel
+---
+
+Coming Soon!
```
Comment (svrnm) [inline content/en/docs/kubernetes/helm.md:7]:
~~~suggestion
You can add the [OpenTelemetry Helm Charts](https://github.com/open-telemetry/opentelemetry-helm-charts) repository using the [Helm](https://helm.sh/) CLI as follows:

```sh
helm repo add open-telemetry https://open-telemetry.github.io/opentelemetry-helm-charts
```
~~~

File: content/en/docs/kubernetes/k8s-operator/target-allocator.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+title: Target Allocator
+weight: 11
+description:
+  A tool to distribute targets of the PrometheusReceiver on all deployed
+  Collector instances
+---
+
+Coming Soon!
```
Comment (svrnm) [inline content/en/docs/kubernetes/k8s-operator/target-allocator.md:9]:
```suggestion
The OpenTelemetry Operator comes with an optional component, the Target Allocator (TA). Currently you can find the documentation for these in the section [Target Allocator in the OpenTelemetry Operator README.md](https://github.com/open-telemetry/opentelemetry-operator#target-allocator)
```

**Comment by svrnm** [comment]
love it! Provided some feedback around the "coming soon" pages: let's remove them or have some pointers to existing documentation.

File: content/en/docs/kubernetes/_index.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
+title: OpenTelemetry with Kubernetes
+linkTitle: Kubernetes
+weight: 11
+description: Using OpenTelemetry with Kubernetes
+spelling: cSpell:ignore Otel
+aliases: [/docs/kubernetes]
```
Comment (chalin) [inline content/en/docs/kubernetes/_index.md:7]:
```suggestion
```

File: content/en/docs/kubernetes/collector.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+title: OpenTelemetry Collector and Kubernetes
+weight: 4
+spelling: cSpell:ignore Otel
```
Comment (chalin) [inline content/en/docs/kubernetes/collector.md:4]:
```suggestion
title: OpenTelemetry Collector and Kubernetes
linkTitle: Collector
```

File: content/en/docs/kubernetes/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+---
+title: Getting Started
+spelling: cSpell:ignore Otel
+weight: 1
```
Comment (chalin) [inline content/en/docs/kubernetes/getting-started.md:4]:
```suggestion
title: Getting Started
weight: -1
```

**Review by chalin — CHANGES_REQUESTED** [review]
Nice! See inline comments.

File: content/en/docs/kubernetes/collector.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+---
+
+Coming Soon!
```
Comment (jpkrohling) [inline content/en/docs/kubernetes/collector.md:7]: If you don't have a content for this already, feel free to get whatever you need from here: https://medium.com/opentelemetry/deploying-the-opentelemetry-collector-on-kubernetes-2256eca569c9

File: content/en/docs/kubernetes/k8s-operator/_index.md [hunk 7]
```diff
@@ -1,13 +1,13 @@
 ---
 title: OpenTelemetry Operator for Kubernetes
 linkTitle: K8s Operator
```
[the remark this answers is not in the document]
Comment (jpkrohling) [reply content/en/docs/kubernetes/k8s-operator/_index.md:3]: Why not Kubernetes Operator? It fits there.

File: content/en/docs/kubernetes/collector.md (hunk 6 above)
Comment (TylerHelmuth) [reply content/en/docs/kubernetes/collector.md]: Thanks! I'll keep that in mind for future PRs.

**Comment by TylerHelmuth** [comment]
I believe I have addressed all feedback

File: content/en/docs/kubernetes/collector.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
+---
+title: OpenTelemetry Collector and Kubernetes
+linkTitle: Collector
+spelling:
+  cSpell:ignore filelogreceiver hostmetricsreceiver kubeletstatsreceiver
+  k8sobjectsreceiver k8sclusterreceiver k8sattributesprocessor
```
Comment (chalin) [inline content/en/docs/kubernetes/collector.md:6]:
```suggestion
  cSpell:ignore k8sobjectsreceiver k8sclusterreceiver k8sattributesprocessor
```

**Review by chalin — COMMENTED** [review]
Looking better!

See suggested inline change to quiet the spelling checker (all local words need to be one the same line as a `cSpell:ignore` directive).

**Review by chalin — APPROVED** [review]

**Comment by cartermp** [comment]
We're doin' it

**Pull request merged**